HumanDynamicsEstimationLibrary/algorithms/src/DynamicalInverseKinematics.cpp

```diff
@@ -0,0 +1,1378 @@
+        pImpl->getTargetRefIfItExists(linkName);
+    if (target == pImpl->m_targets.end()) {
+        return false;
+    }
+
+    target->second.setOrientation(orientation);
+    target->second.setAngularVelocity(angularVelocity);
+    return true;
+}
+
+bool DynamicalInverseKinematics::updateTargetLinearVelocity(const std::string& linkName,
+                                                            const iDynTree::Vector3& linearVelocity)
+{
+    DynamicalInverseKinematics::impl::TargetsMap::iterator target =
+        pImpl->getTargetRefIfItExists(linkName);
+    if (target == pImpl->m_targets.end()) {
+        return false;
+    }
+
+    target->second.setLinearVelocity(linearVelocity);
+    return true;
+}
+
+bool DynamicalInverseKinematics::updateTargetAngularVelocity(const std::string& linkName,
+                                                             const iDynTree::Vector3& angularVelocity)
+{
+    DynamicalInverseKinematics::impl::TargetsMap::iterator target =
+        pImpl->getTargetRefIfItExists(linkName);
+    if (target == pImpl->m_targets.end()) {
+        return false;
+    }
+
+    target->second.setAngularVelocity(angularVelocity);
+    return true;
+}
+
+bool DynamicalInverseKinematics::updateTargetGains(const std::string& linkName,
+                                                   const double positionFeedbackGain,
+                                                   const double orientationFeedbackGain,
+                                                   const double linearVelocityFeedforwardGain,
+                                                   const double angularVelocityFeedforwardGain)
+{
+    DynamicalInverseKinematics::impl::TargetsMap::iterator target =
+        pImpl->getTargetRefIfItExists(linkName);
+    if (target == pImpl->m_targets.end()) {
+        return false;
+    }
+
+    target->second.setPositionFeedbackGain(positionFeedbackGain);
+    target->second.setOrientationFeedbackGain(orientationFeedbackGain);
+    target->second.setLinearVelocityFeedforwardGain(linearVelocityFeedforwardGain);
+    target->second.setAngularVelocityFeedforwardGain(angularVelocityFeedforwardGain);
+    return true;
+}
+
+bool DynamicalInverseKinematics::updateTargetWeights(const std::string& linkName,
+                                                   const double linearVelocityWeight,
+                                                   const double angularVelocityWeight)
+{
+    DynamicalInverseKinematics::impl::TargetsMap::iterator target =
+        pImpl->getTargetRefIfItExists(linkName);
+    if (target == pImpl->m_targets.end()) {
+        return false;
+    }
+
+    target->second.setLinearVelocityWeight(linearVelocityWeight);
+    target->second.setAngularVelocityWeight(angularVelocityWeight);
+    return true;
+}
+
+
+
+bool DynamicalInverseKinematics::getJointsConfigurationSolution(iDynTree::VectorDynSize& jointsConfiguration) const
+{
+    if (jointsConfiguration.size() == pImpl->m_state.s.size()) {
```

Comment:
I normally write these checks in methods like this: 
```
if(!condition)
{return false;}

do operation;
return true;
````
However, it is a design choice for sure. I am not sure if there are some standard practices for that!

Reply:
fixed in https://github.com/robotology/human-dynamics-estimation/pull/265/commits/464177c5d794cb35c8739bdf76a23de82ed1cc2d